hbase-server/src/test/java/org/apache/hadoop/hbase/master/TestMasterOperationsForRegionReplicas.java

```diff
@@ -192,86 +202,95 @@ public void testCreateTableWithMultipleReplicas() throws Exception {
+        numReplica, regions.size() == numRegions * (numReplica + 1));
 
-      //decrease the replica(earlier, table was modified to have a replica count of numReplica + 1)
+      // decrease the replica(earlier, table was modified to have a replica count of numReplica + 1)
       ADMIN.disableTable(tableName);
-      desc.setRegionReplication(numReplica);
-      ADMIN.modifyTable(desc);
+      ADMIN.modifyTable(
+        TableDescriptorBuilder.newBuilder(desc).setRegionReplication(numReplica).build());
       ADMIN.enableTable(tableName);
-      assert(ADMIN.isTableEnabled(tableName));
-      regions = TEST_UTIL.getMiniHBaseCluster().getMaster()
-          .getAssignmentManager().getRegionStates().getRegionsOfTable(tableName);
-      assert(regions.size() == numRegions * numReplica);
-      //also make sure the meta table has the replica locations removed
+      assertTrue(ADMIN.isTableEnabled(tableName));
+      TEST_UTIL.waitUntilAllRegionsAssigned(tableName);
+      TEST_UTIL.waitUntilNoRegionsInTransition();
+      regions = TEST_UTIL.getMiniHBaseCluster().getMaster().getAssignmentManager().getRegionStates()
+        .getRegionsOfTable(tableName);
+      assertEquals(numRegions * numReplica, regions.size());
+      // also make sure the meta table has the replica locations removed
       hris = MetaTableAccessor.getTableRegions(ADMIN.getConnection(), tableName);
-      assert(hris.size() == numRegions * numReplica);
-      //just check that the number of default replica regions in the meta table are the same
-      //as the number of regions the table was created with, and the count of the
-      //replicas is numReplica for each region
+      assertEquals(numRegions * numReplica, hris.size());
+      // just check that the number of default replica regions in the meta table are the same
+      // as the number of regions the table was created with, and the count of the
+      // replicas is numReplica for each region
       Map<RegionInfo, Integer> defaultReplicas = new HashMap<>();
       for (RegionInfo hri : hris) {
         Integer i;
         RegionInfo regionReplica0 = RegionReplicaUtil.getRegionInfoForDefaultReplica(hri);
         defaultReplicas.put(regionReplica0,
-            (i = defaultReplicas.get(regionReplica0)) == null ? 1 : i + 1);
+          (i = defaultReplicas.get(regionReplica0)) == null ? 1 : i + 1);
       }
-      assert(defaultReplicas.size() == numRegions);
+      assertEquals(numRegions, defaultReplicas.size());
       Collection<Integer> counts = new HashSet<>(defaultReplicas.values());
-      assert(counts.size() == 1 && counts.contains(numReplica));
+      assertEquals(1, counts.size());
+      assertTrue(counts.contains(numReplica));
     } finally {
       ADMIN.disableTable(tableName);
       ADMIN.deleteTable(tableName);
     }
   }
 
-  @Test @Ignore("Enable when we have support for alter_table- HBASE-10361")
+  @Test
+  @Ignore("Enable when we have support for alter_table- HBASE-10361")
```

Comment:
Unrelated, but HBASE-10361 has merged ages ago. I will open a JIRA to remove @Ignore.

Reply:
Let me include this in the new patch, just a one line change.

Reply:
Oh the test fails locally for me... Then let's use a new issue to address it...

Reply:
It also fails for me and the try block catches the underlying exception from `ADMIN.enableTable(tableName);`. Better to work on this on a new issue.